staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -176,7 +180,22 @@ func (q *delayingType) AddAfter(item interface{}, duration time.Duration) {
 	select {
```

Comment:
reagarding the `q.metrics.retry()` in `AddAfter`. It depends what we want to measure with it.

We advertize this:

https://github.com/kubernetes/kubernetes/blob/d62cc3dc6d5c07fea79eafd866ac7e1217000ea8/staging/src/k8s.io/component-base/metrics/prometheus/workqueue/metrics.go#L93

But currently, it only correctly measures calls to `AddAfter`.  It might not correspond to the number of adds to the workqueue  since it also measures retries of the same element that is still waiting (will be queued only once).

This metric might be even less precise when used together with  `ForgetDelayed`, since it will also count items that were canceled.

We could also decide to trigger the metric just before each `q.Add(item)` statement to be sure what we track. Nevertheless it will change the original behaviour.

Thoughts?